library/src/amd_detail/hipblas.cpp

```diff
@@ -13992,6 +13993,489 @@ catch(...)
     return exception_to_hipblas_status();
 }
 
+// trmmOutofplace
+hipblasStatus_t hipblasStrmmOutofplace(hipblasHandle_t    handle,
+                                       hipblasSideMode_t  side,
+                                       hipblasFillMode_t  uplo,
+                                       hipblasOperation_t transA,
+                                       hipblasDiagType_t  diag,
+                                       int                m,
+                                       int                n,
+                                       const float*       alpha,
+                                       const float*       A,
+                                       int                lda,
+                                       const float*       B,
+                                       int                ldb,
+                                       float*             C,
+                                       int                ldc)
+try
+{
+    return rocBLASStatusToHIPStatus(rocblas_strmm_outofplace((rocblas_handle)handle,
+                                                             hipSideToHCCSide(side),
+                                                             hipFillToHCCFill(uplo),
+                                                             hipOperationToHCCOperation(transA),
+                                                             hipDiagonalToHCCDiagonal(diag),
+                                                             m,
+                                                             n,
+                                                             alpha,
+                                                             A,
+                                                             lda,
+                                                             B,
+                                                             ldb,
+                                                             C,
+                                                             ldc));
```

Comment:
You don't want to define ROCBLAS_V3 and test that system here and use the new name?

Reply:
I think it is works as it is, and it seems to me to be more similar in amd_detail and nvidia_detail. An alternative would be to define ROCBLAS_V3 and use the new name. Below is how it works as it is: 

The librocblas.so contains both rocblas_strmm(... A, lda, B, ldb...) and rocblas_strmm_outofplace(...A, lda, B, ldb, C, ldc ...). 

Same for libhipblas.so, it contains both hipblasStrmm( ... A, lda, B, ldb, ...) and hipblasStrmmOutofplace(... A, lda, B, ldb, C, ldc, ...). 

If you compile without HIPBLAS_V1 you can call either hipblasStrmm or hipblasStrmmOutofplace . You get hipBLAS deprecation messages for both hipblasStrmm and hipblasStrmmOutofplace (rocBLAS deprecation messages are turned off at line 23 with # define ROCBLAS_NO_DEPRECATION_WARNINGS)

If you compile with HIPBLAS_V1 then any call you make to hipblasStrmm(  A, lda, B, ldb, C, ldc, ...) in your code is converted to a call to hipblasStrmmOutofplace(..A, lda, B, ldb, C, ldc)  by cpp. Calls to hipblasStrmm(.. A, lda, B, ldb ...) will give link errors because they are converted to hipblasStrmmOutofplace( ... A, lda, B, ldb, C, ldc ...) and there is an argument mismatch.